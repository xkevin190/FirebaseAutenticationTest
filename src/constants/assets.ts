export const FONTS = {
    LATO: 'Lato',
};