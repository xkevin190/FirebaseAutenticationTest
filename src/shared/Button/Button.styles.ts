import {StyleSheet, ViewStyle} from 'react-native';

const basicContainer: ViewStyle = {
  justifyContent: 'center',
  alignItems: 'center',
  height: 50,
  borderRadius: 25,
};

const styles = StyleSheet.create({
  md: {
    ...basicContainer,
    width: '45%',
  } as ViewStyle,
  xxl: {
    ...basicContainer,
    width: '100%',
  } as ViewStyle,
});

export default styles;
