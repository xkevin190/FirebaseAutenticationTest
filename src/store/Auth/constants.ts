export type IAccountActionTypes = {
  CREATE_ACCOUNT: string;
  LOGIN: string;
  ClOSE_SESSION: string;
};

export const AccounthActionsTypes: IAccountActionTypes = {
  CREATE_ACCOUNT: 'CREATE_ACCOUNT',
  LOGIN: ' LOGIN',
  ClOSE_SESSION: 'ClOSE_SESSION',
};
