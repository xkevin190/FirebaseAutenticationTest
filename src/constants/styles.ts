import { FONTS } from "./assets";

export const COLORS = {
    WHITE: '#FFFFFF',
    BLACK: '#000000',
    BLACK_025: '#00000040',
    GRAY_DEFAULT: '#46484B',
    GRAY_DEFAULT_DARKER: '#646565',
    GRAY_MEDIUM: '#9DA5B9',
    GRAY_LIGHT: '#F6F8FF',
    GRAY_LIGHT_DARKER: '#DDDDDD',
    DISABLED_DEFAULT: '#DFE6ED',
    OPYA_BLUE: '#2162E3',
    OPYA_VIOLET: '#1745D5',
    MAIN_DEFAULT: '#313342',
    SECONDARY_LIGHT: '#62698B',
    ALERT_DEFAULT: '#D15686',
    ALERT_LIGHT: '#FFD0E2',
  };

  const FONTS_TYPES = {
    BLACK: 'Black',
    BLACK_ITALIC: 'BlackItalic',
    BOLD: 'Bold',
    BOLD_ITALIC: 'BoldItalic',
    HAIR_LINE: 'HairLine',
    HAIR_LINE_ITALIC: 'HairLineItalic',
    HEAVY: 'Heavy',
    HEAVY_ITALIC: 'HeavyItalic',
    ITALIC: 'Italic',
    LIGHT: 'Light',
    LIGHT_ITALIC: 'LightItalic',
    MEDIUM: 'Medium',
    MEDIUM_ITALIC: 'MediumItalic',
    REGULAR: 'Regular',
    SEMI_BOLD: 'SemiBold',
    SEMI_BOLD_ITALIC: 'SemiBoldItalic',
    THIN: 'Thin',
    THIN_ITALIC: 'ThinItalic',
  };


  export const STYLES = {
    FONT_FAMILY: {
      LATO_REGULAR: `${FONTS.LATO}-${FONTS_TYPES.REGULAR}`,
      LATO_BOLD: `${FONTS.LATO}-${FONTS_TYPES.BOLD}`,
    },
    FONT_SIZE: {
      PARAGRAPH: 14,
      SMALL_PARAGRAPH: 12,
      MEDIUM_PARAGRAPH: 16,
      INTERMEDIUM_PARAGRAPH: 18,
      LARGE_PARAGRAPH: 20,
    },
  };
  

