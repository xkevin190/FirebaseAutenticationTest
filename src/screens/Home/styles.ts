import {StyleSheet, ViewStyle} from 'react-native';

import {COLORS} from '../../constants/styles';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'column',
    backgroundColor: COLORS.GRAY_LIGHT,
  } as ViewStyle,
});

export default styles;
