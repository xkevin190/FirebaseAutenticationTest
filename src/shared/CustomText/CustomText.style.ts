import {StyleSheet, TextStyle} from 'react-native';

import {STYLES} from '../../constants/styles';

const styles = StyleSheet.create({
  'small-paragraph-bold': {
    fontFamily: STYLES.FONT_FAMILY.LATO_BOLD,
    fontSize: STYLES.FONT_SIZE.SMALL_PARAGRAPH,
  } as TextStyle,
  'small-paragraph': {
    fontFamily: STYLES.FONT_FAMILY.LATO_REGULAR,
    fontSize: STYLES.FONT_SIZE.SMALL_PARAGRAPH,
  } as TextStyle,
  'medium-paragraph': {
    fontFamily: STYLES.FONT_FAMILY.LATO_REGULAR,
    fontSize: STYLES.FONT_SIZE.MEDIUM_PARAGRAPH,
  } as TextStyle,
  paragraph: {
    fontFamily: STYLES.FONT_FAMILY.LATO_REGULAR,
    fontSize: STYLES.FONT_SIZE.PARAGRAPH,
  } as TextStyle,
  'large-paragraph': {
    fontFamily: STYLES.FONT_FAMILY.LATO_REGULAR,
    fontSize: STYLES.FONT_SIZE.LARGE_PARAGRAPH,
  } as TextStyle,
  'paragraph-bold': {
    fontFamily: STYLES.FONT_FAMILY.LATO_BOLD,
    fontSize: STYLES.FONT_SIZE.PARAGRAPH,
  } as TextStyle,
  'intermedium-paragraph-bold': {
    fontFamily: STYLES.FONT_FAMILY.LATO_BOLD,
    fontSize: STYLES.FONT_SIZE.INTERMEDIUM_PARAGRAPH,
  } as TextStyle,
  'intermedium-paragraph': {
    fontSize: STYLES.FONT_SIZE.INTERMEDIUM_PARAGRAPH,
    fontFamily: STYLES.FONT_FAMILY.LATO_REGULAR,
  } as TextStyle,
});

export default styles;
