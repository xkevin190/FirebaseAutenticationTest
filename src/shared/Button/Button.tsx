import React from 'react';
import { TouchableOpacity } from 'react-native';

import { ButtonSize } from '../../types/styles';
import CustomText from '../CustomText';
import styles from './Button.styles';
import { COLORS } from '../../constants/styles';

export interface IButtonProps {
  buttonText: string;
  testID: string;
  size?: ButtonSize;
  rounded?: boolean;
  color?: string;
  textColor?: string;
  disabled?: boolean;
  onPress: () => void;
}

const Button: React.FC<IButtonProps> = ({
  buttonText,
  testID,
  size = 'md',
  onPress,
  color = COLORS.OPYA_BLUE,
  rounded = false,
  textColor = undefined,
  disabled = false,
}) => (
  <TouchableOpacity
    style={{
      ...styles[size],
      backgroundColor: disabled ? COLORS.DISABLED_DEFAULT : color,
      ...(!rounded && { borderRadius: 0 }),
    }}
    onPress={onPress}
    testID={testID}
    disabled={disabled}
  >
    <CustomText text={buttonText} color={disabled ? COLORS.BLACK : textColor} />
  </TouchableOpacity>
);

export default Button;
