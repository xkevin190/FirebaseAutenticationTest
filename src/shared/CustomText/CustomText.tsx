import React from 'react';
import {Text, TextStyle} from 'react-native';

import {COLORS} from '../../constants/styles';
import {TextBody} from '../../types/CustomText';
import styles from './CustomText.style';

export interface ICustomTextProps {
  text: string;
  center?: boolean;
  color?: string;
  bodyType?: TextBody;
  testID?: string;
  customStyle?: TextStyle;
}

const CustomText: React.FC<ICustomTextProps> = ({
  text,
  color = COLORS.BLACK,
  bodyType = 'paragraph',
  center = false,
  testID = '',
  customStyle = {},
}) => (
  <Text
    style={{
      ...styles[bodyType],
      ...customStyle,
      ...(center && {textAlign: 'center'}),
      color,
    }}
    testID={testID}>
    {text}
  </Text>
);

export default CustomText;
