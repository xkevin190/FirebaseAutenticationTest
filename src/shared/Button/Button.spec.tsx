import React from 'react';
import { TouchableOpacityProps } from 'react-native';
import { shallow, ShallowWrapper } from 'enzyme';

import styles from './Button.styles';
import { ICustomTextProps } from '../CustomText/CustomText';
import { COLORS } from '../../constants/styles';
import Button, { IButtonProps } from './Button';

describe('Button (Unit-Test)', () => {
  const props: IButtonProps = {
    buttonText: 'BUTTON_TEXT',
    onPress: jest.fn(),
    testID: 'BUTTON_TEST_ID',
  };
  let wrapper: ShallowWrapper;

  describe('general functionality', () => {
    beforeEach(() => {
      wrapper = shallow(<Button {...props} />);
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should render "buttonText" prop', () => {
      const textProps = wrapper.find('CustomText').props() as ICustomTextProps;
      expect(textProps.text).toEqual(props.buttonText);
    });

    it('should render default button style', () => {
      const buttonProps = wrapper
        .find('TouchableOpacity')
        .props() as TouchableOpacityProps;
      expect(buttonProps.style).toEqual({
        ...styles.md,
        backgroundColor: COLORS.OPYA_BLUE,
        borderRadius: 0,
      });
    });

    it('should set "testID" prop received to "testID" prop of button', () => {
      const buttonProps = wrapper
        .find('TouchableOpacity')
        .props() as TouchableOpacityProps;
      expect(buttonProps.testID).toEqual(props.testID);
    });

    it('should call "onPress" prop when tapped', () => {
      const button = wrapper.find('TouchableOpacity');
      button.simulate('press');
      expect(props.onPress).toBeCalledTimes(1);
    });
  });

  describe('when optional props are received', () => {
    it('should render button background color with "color" prop received', () => {
      wrapper = shallow(<Button {...props} color={COLORS.BLACK} />);
      const buttonProps = wrapper
        .find('TouchableOpacity')
        .props() as TouchableOpacityProps;
      expect(buttonProps.style).toEqual({
        ...styles.md,
        backgroundColor: COLORS.BLACK,
        borderRadius: 0,
      });
    });

    it('should render button borderRadius when "rounded" prop received', () => {
      wrapper = shallow(<Button {...props} rounded />);
      const buttonProps = wrapper
        .find('TouchableOpacity')
        .props() as TouchableOpacityProps;
      expect(buttonProps.style).toEqual({
        ...styles.md,
        backgroundColor: COLORS.OPYA_BLUE,
        borderRadius: 25,
      });
    });

    it('should render "buttonText" with "textColor" prop received', () => {
      wrapper = shallow(<Button {...props} textColor={COLORS.GRAY_DEFAULT} />);
      const textProps = wrapper.find('CustomText').props() as ICustomTextProps;
      expect(textProps.color).toEqual(COLORS.GRAY_DEFAULT);
    });

    it('should render disabled button when "disabled" prop its received', () => {
      wrapper = shallow(<Button {...props} disabled />);
      const textProps = wrapper.find('CustomText').props() as ICustomTextProps;
      const buttonProps = wrapper
        .find('TouchableOpacity')
        .props() as TouchableOpacityProps;
      expect(buttonProps.disabled).toEqual(true);
      expect(buttonProps.style).toEqual({
        ...styles.md,
        borderRadius: 0,
        backgroundColor: COLORS.DISABLED_DEFAULT,
      });
      expect(textProps.color).toEqual(COLORS.BLACK);
    });
  });
});
