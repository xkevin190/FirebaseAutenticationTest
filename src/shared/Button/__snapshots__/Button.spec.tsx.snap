// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button (Unit-Test) general functionality should match snapshot 1`] = `
<TouchableOpacity
  disabled={false}
  onPress={[MockFunction]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#2162E3",
      "borderRadius": 0,
      "height": 50,
      "justifyContent": "center",
      "width": "45%",
    }
  }
  testID="BUTTON_TEST_ID"
>
  <CustomText
    text="BUTTON_TEXT"
  />
</TouchableOpacity>
`;
